### ISSUE TITLE
paddlepaddle 0.10.0运行出core

### ISSUE THREAD

**Issue by xieshufu** [PaddlePaddle/Paddle#3723]
运行后出core，相关信息如下：
```text 
I0828 20:04:42.480975 13487 Trainer.cpp:165] trainer mode: Normal
*** Aborted at 1503921882 (unix time) try "date -d @1503921882" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGFPE (@0x7fde055584f2) received by PID 13487 (TID 0x7fde1ea33860) from PID 89490674; stack trace: ***
    @       0x318b20f500 (unknown)
    @     0x7fde055584f2 longdouble_multiply
    @     0x7fde1fc33962 binary_op1
    @     0x7fde1fc350b8 PyNumber_Multiply
    @     0x7fde1fcc86b9 PyEval_EvalFrameEx
    @     0x7fde1fcce120 PyEval_EvalCodeEx
    @     0x7fde1fcce232 PyEval_EvalCode
    @     0x7fde1fcdd752 PyImport_ExecCodeModuleEx
    @     0x7fde1fcddbf6 load_source_module
    @     0x7fde1fcdf07f import_submodule
    @     0x7fde1fcdf5dc ensure_fromlist
    @     0x7fde1fcdfbbf import_module_level
    @     0x7fde1fce06b4 PyImport_ImportModuleLevel
    @     0x7fde1fcc5c9f builtin___import__
    @     0x7fde1fc320e3 PyObject_Call
    @     0x7fde1fcc6183 PyEval_CallObjectWithKeywords
    @     0x7fde1fcc8c73 PyEval_EvalFrameEx
    @     0x7fde1fcce120 PyEval_EvalCodeEx
    @     0x7fde1fcce232 PyEval_EvalCode
    @     0x7fde1fcdd752 PyImport_ExecCodeModuleEx
    @     0x7fde1fcddbf6 load_source_module
    @     0x7fde1fcde98d load_package
    @     0x7fde1fcdf07f import_submodule
    @     0x7fde1fcdf314 load_next
    @     0x7fde1fcdf988 import_module_level
    @     0x7fde1fce06b4 PyImport_ImportModuleLevel
    @     0x7fde1fcc5c9f builtin___import__
    @     0x7fde1fc320e3 PyObject_Call
    @     0x7fde1fcc6183 PyEval_CallObjectWithKeywords
    @     0x7fde1fcc8c73 PyEval_EvalFrameEx
    @     0x7fde1fcce120 PyEval_EvalCodeEx
    @     0x7fde1fcce232 PyEval_EvalCode
/home/vis/.jumbo/bin/paddle: line 96: 13487 Floating point exception(core dumped) ${DEBUGGER} $PADDLE_BIN_PATH/paddle_trainer ${@:2}

/home/vis/.jumbo/bin/paddle: line 96的信息如下：
 94 
 95 
 96 case "$1" in
 97     "train")
 98         ${DEBUGGER} $PADDLE_BIN_PATH/paddle_trainer ${@:2}
 99         ;;
100     "merge_model")
101         ${DEBUGGER} $PADDLE_BIN_PATH/paddle_merge_model ${@:2}
102         ;;
103     "pserver")
104         ${DEBUGGER} $PADDLE_BIN_PATH/paddle_pserver_main ${@:2}
105         ;;
file `which python`的相关信息如下：
/home/vis/.jumbo/bin/python: symbolic link to `python2'
file /home/vis/.jumbo/bin/python2.7的相关信息如下：
/home/vis/.jumbo/bin/python2.7: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.18, not stripped
```

**Comment by typhoonzero** [comment]
可以贴一下模型的配置么？

**Comment by xieshufu** [comment]
模型结构大概如下：
#edit-mode: -*- python -*-
#coding:utf-8
#from paddle.trainer_config_helpers import *
trainer_id = get_config_arg('trainer_id', int, 0)
local = get_config_arg('local', bool, True)

TrainData(PyData(
    files = "/home/vis/liushanshan/english/model/8conv_203w_added_filter_vir_symbol/train_list/list_train_set.txt",
    load_data_module="pyImageDataProvider_ctc_batch",
    load_data_object="SimpleDataProvider",
    async_load_data=True 
))

TestData(PyData(
    files = "/home/vis/liushanshan/english/model/8conv_203w_added_filter_vir_symbol/train_list/list_test_set.txt",
    load_data_module="pyImageDataProvider_test_ctc_batch",
    load_data_object="SimpleDataProvider_test"
))

#default_gradient_clipping_threshold(10)
Settings(
    algorithm='sgd',
    average_window=0.15,
    batch_size=32,
    learning_rate=0.1
    )

Inputs("input", "label")
Outputs("cost")

Layer(
    name = "input",
    type = "data",
    size = 2304,
)

Layer(
    name = "conv1_1",
    type = "exconv",
    device = 0,
    active_type = "relu",
    bias = Bias(learning_rate=0.001,
        momentum=0.9,
        initial_mean=0,
        initial_std=0.0),
    inputs = Input("input",
        learning_rate=0.001,
        momentum=0.9,
        decay_rate=0.0004,
        initial_mean=0.0,
        initial_std=0.0005,
        conv = Conv(filter_size = 3,
             channels = 1,
             padding = 1,
             stride = 1,
             groups = 1)),
    num_filters = 16,
    #partial_sum = 1,
    shared_biases = True
)
Layer(
    name = "conv1_2",
    type = "exconv",
    device = 0,
    active_type = "relu",
    bias = Bias(learning_rate=0.001,
        momentum=0.9,
        initial_mean=0,
        initial_std=0.0),
    inputs = Input("conv1_1",
        learning_rate=0.001,
        momentum=0.9,
        decay_rate=0.0004,
        initial_mean=0.0,
        initial_std=0.01,
        conv = Conv(filter_size = 3,
             channels = 16,
             padding = 1,
             stride = 1,
             groups = 1)),
    num_filters = 16,
    #partial_sum = 1,
    shared_biases = True
)

Layer(
    name = "pool1",
    type = "pool",
    device = 0,
    inputs = Input("conv1_2",
        pool = Pool(pool_type="max-projection",
            channels = 16,
            size_x = 2,
            start = 0,
            stride = 2))
)
... ...
Layer(
    name = "output",
    type = "fc",
    device = 0,
    size = 96,
    #active_type = "softmax",
    bias = Bias(learning_rate=0.001,
        momentum=0.9,
        initial_mean=0,
        initial_std=0),
    inputs = [Input("gated1",
        learning_rate=0.001,
        momentum=0.9,
        decay_rate=0.0004,
        initial_mean=0.0,
        initial_std=0.02),
              Input("gated2",
        learning_rate=0.001,
        momentum=0.9,
        decay_rate=0.0004,
        initial_mean=0.0,
        initial_std=0.02)],
)

Layer(
    name = "label",
    type = "data",
    device = 0,
    size = 0,
)

Layer(
    name = "cost",
    type = "warp_ctc",
    device = 0,
    size = 96,
    norm_by_times = True,
    blank = 95,
    inputs = ["output", "label"],
)

Evaluator(
    name = "error",
    type = "ctc_edit_distance",
    inputs = ["output", "label"],
)

**Comment by qingqing01** [comment]
> SIGFPE (@0x7fde055584f2) received by PID 13487 (TID 0x7fde1ea33860) from PID 89490674; stack trace: ***

SIGFPE是浮点异常错误，可以调下参数试下~

**Comment by xieshufu** [comment]
没有解决啊，修改learning rate也不能解决，感觉是系统环境配置的问题。

**Comment by hedaoyuan** [comment]
> 没有解决啊，修改learning rate也不能解决，感觉是系统环境配置的问题。

@xieshufu 我reopen了这个issue，但是问题原因稍微描述详细一些吧，基于这样的描述基本很难猜出来具体原因。

**Comment by xieshufu** [comment]
这个问题是这样： 在一台GPU机器上，安装了PADDLE后，只要运行就出这个core。而在另外的GPU机器上，运行PADDLE是正常的。怀疑是出问题机器上的python等版本有问题，但后来查过，也没有解决。

**Comment by hedaoyuan** [comment]
@xieshufu 你上面贴的信息比一开始贴的那段调用栈可能更接近原因。
> 在一台GPU机器上，安装了PADDLE后，只要运行就出这个core。而在另外的GPU机器上，运行PADDLE是正常的。

我觉得，比较简单的验证方式，可以使用book中的一个简单例子（比如recognize_digits那个）试一下是否在你出问题的机器上是可以正常运行paddle的。

**Comment by xieshufu** [comment]
再补充一个信息，在有问题的这台机器上，跑PADDLE v2的训练是可以的，跑旧版本的训练程序会出这个问题。 这个旧版本大概是今年初的一个版本。

**Comment by Yipeng-Sun** [comment]
可能与环境和相关编译依赖有关

**Comment by xieshufu** [comment]
后来确认后发现numpy版本的问题， 正常运行机器的numpy版本是： numpy (1.12.0)；出问题机器上的numpy版本是: 1.13.1, 将numpy卸载掉后重新安装指定版本的numpy(1.12.0)，问题就解决了。

**Comment by typhoonzero** [comment]
我在我的mac笔记本上也复现了这个问题，但降级到`numpy==1.12`后运行v1会报编译的二进制是基于`1.13`的，仍然不能执行。需要修复一下。
